pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/JavaQualifiedName.java

```diff
@@ -115,9 +115,24 @@ private static JavaQualifiedName ofOperation(JavaQualifiedName parent, String op
     }
 
 
-    // Might be useful with type resolution
+    /**
+     * Gets the qualified name of a class.
+     *
+     * @param clazz Class object
+     *
+     * @return The qualified name of the class, or null if the class is null
+     */
     public static JavaQualifiedName ofClass(Class<?> clazz) {
-        throw new UnsupportedOperationException();
+        if (clazz == null) {
+            return null;
+        }
+
+        String name = clazz.getName();
+        if (name.indexOf('.') < 0) {
+            name = '.' + name;
```

Comment:
This looks a bit strange, why to prefix with a dot... but that's the format for ofString when using the default package... Maybe just add a little comment here "// unnamed package, preceded by a full stop. See ofString.